scripts/linters/pre_commit_linter.py

```diff
@@ -498,10 +498,14 @@ def main(args=None):
     all_messages = []
 
     for task in tasks_custom:
+        custom_semaphore.acquire()
```

Comment:
You'd still want to use a distinct semaphore for this, the value would be 1 in this case.

Reply:
@kevinlee12 do I need to need to use separate semaphore for custom and third party tasks or a single semaphore does the work?

Reply:
No need, you just need to use a distinct semaphore like what did when you had the lock. (They'll serve the same purpose).

Reply:
Done.